lib/elixir/src/elixir_erl.erl

```diff
@@ -559,6 +581,38 @@ get_type_docs(Set, Types) ->
      {_, Line, Doc} <- ets:lookup(Set, {Kind, Tuple})]
   ).
 
+translate_doc({{Name, Arity}, Line, Kind, Signature, Doc}) ->
+  {{translate_doc_kind(Kind), Name, Arity},
+   erl_anno:new(Line),
+   [signature_to_binary(Name, Signature)],
+   doc_value(Doc),
+   #{signature => Signature}
+  };
+translate_doc({{Name, Arity}, Line, Kind, Doc}) ->
+  {{translate_doc_kind(Kind), Name, Arity},
+   erl_anno:new(Line),
+   [],
+   doc_value(Doc),
+   #{}
+  }.
+
+translate_doc_kind(def)           -> function;
+translate_doc_kind(callback)      -> callback;
+translate_doc_kind(type)          -> type;
+translate_doc_kind(opaque)        -> opaque;
```

Comment:
I am thinking it may be better to go with `type` and put `opaque` in the metadata. Thoughts?

Reply:
Sounds good to me, I'll do the change.